non-wheel/httpsrv/httpsrv.py

```diff
@@ -0,0 +1,513 @@
+        index_file.write(get_index_html(name))
+
+        with open(path + '/conf/fastcgi.conf', "w+") as file:
+            file.write(get_fastcgi_conf(name))
+        with open(path + '/conf/fastcgi.conf.twn', "w+") as file:
+            file.write(get_fastcgi_conf_twn(name))
+        with open(path + '/conf/fastcgi_dynamic.conf', "w+") as file:
+            file.write(get_fastcgi_dynamic_conf(name))
+        with open(path + '/conf/fastcgi_http_add.conf', "w+") as file:
+            file.write(get_fastcgi_http_add_conf(name))
+
+        get_conf = globals()['get_{}_conf'.format(conf)]
+        with open(path + '/conf/httpd.conf', "w+") as conf_file:
+            conf_file.write(get_conf(name, port))
+
+        os.mkdir(path + '/conf/apache-sites')
+        os.mkdir(path + '/example-project')
+        os.mkdir(path + '/example-project/conf')
+        with open("{}/example-project/conf/{}.conf".format(path, name), "w+") as file:
+            file.write(get_example_vhost_conf(name, port))
+        os.mkdir(path + '/example-project/htdocs')
+        with open(path + '/example-project/htdocs/index.html', "w+") as file:
+            file.write(get_index_html('Example Project'))
+        os.mkdir(path + '/example-project/logs')
+        os.symlink(
+            "{}/example-project/conf/{}.conf".format(path, name),
+            path + '/conf/apache-sites/example-project.conf'
+        )
+
+        # create_with_qzui_api(name)
+        create_with_sql(name)
+
+
+# Hopefully to go away in favor of create_with_qzui_api
+def create_with_sql(name):
+    # This connection is for the SQL way of creating an HTTP Server Instance.
+    conn = db2.connect()
+    cur = conn.cursor()
+
+    sys_cmd = "CPYF FROMFILE(QUSRSYS/QATMHINSTC) TOFILE(QUSRSYS/QATMHINSTC) " +\
+        "FROMMBR(APACHEDFT) TOMBR({}) MBROPT(*REPLACE)".format(name)
+    crt_http_svr = "call qcmdexc('{}');".format(sys_cmd)
+    cur.execute(crt_http_svr)
+
+    query = 'create or replace alias QUSRSYS.QATMHINSTC_{0} for QUSRSYS.QATMHINSTC({0})'.format(name)
+    cur.execute(query)
+
+    query = '''update QUSRSYS.QATMHINSTC_{0}
+set CHARFIELD = '-apache -d /www/{0} -f conf/httpd.conf -AutoStartN' with NC'''.format(name)
+    cur.execute(query)
+
+    conn.commit()
+    cur.close()
+    conn.close()
+
+
+# This is the part that needs a lot of help.
+# I (Josh) do not know the Python Toolkit for IBM i. I'm willing to learn.
+def create_with_qzui_api(name):
+    itransport = iLibCall()
+    itool = iToolKit()
+
+    itool.add(iCmd('addlible', 'addlible QHTTPSVR'))
+    itool.add(
+        iSrvPgm('QzuiCreateInstance', 'QZHBCONF', 'QzuiCreateInstance',
+                iopt={'lib': 'QHTTPSVR'})  # This doesn't seem to work, thus the ADDLIBLE above
+            .addParm(iData('instance', '10a', name.upper()))
+            .addParm(
+                iDS('INSD0110', {'len': 'buflen'})
+                    .addData(iData('autostart', '10a', '*NO'))
```

Comment:
Looking at the QHTTPSVR/H(QZHBCONF) header file, it seems the structure is not packed, so we need to add 2 bytes of padding here so that the integers following are aligned on a 4-byte boundary (offset 12 instead of 10).

```python
                    .addData(iData('pad_for_alignment', '2h', '0'))
```